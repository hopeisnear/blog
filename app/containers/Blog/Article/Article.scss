@import "app/styles/variables.scss";

.blog-article {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  padding-bottom: 60px;

  a {
    text-decoration: underline;
    color: $active-link-color;
  }

  .blog-article-content {
    margin-bottom: 60px;
  }

  .comment-form-header {
    border: 0;
    margin-bottom: 0;
  }
}
