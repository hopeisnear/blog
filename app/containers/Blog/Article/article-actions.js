import { fetchArticle } from 'services/article-service';

export const FETCH_ARTICLE_RESPONDED = 'FETCH_ARTICLE_RESPONDED';

export function fetchArticleAction(articleName) {
  return (dispatch) => fetchArticle(articleName)
    .then((article) => {
      dispatch({ type: FETCH_ARTICLE_RESPONDED, article });
    });
}
