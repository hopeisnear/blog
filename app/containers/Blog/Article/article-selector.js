import { createSelector } from 'reselect';
import { selectArticleName } from '../article-name-selector';

const selectArticle = () => createSelector(
  (state) => state.getIn(['blog', 'articles']),
  selectArticleName(),
  (articles, articleName) => {
    const containsArticle = articles.some((article) => article.get('name') === articleName && article.has('details'));
    return containsArticle ? articles.find((article) => article.get('name') === articleName).get('details') : undefined;
  }
);

export { selectArticle, selectArticleName };
