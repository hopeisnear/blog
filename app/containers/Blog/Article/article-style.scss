.article-wrapper {
  flex-basis: 100%;
  display: flex;
  margin-bottom: 35px;
  overflow: hidden;
  padding: 0 20px;
  .article {
    display: flex;
    flex-direction: column;
  }
}
