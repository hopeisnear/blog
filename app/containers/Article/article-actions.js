import { fetchArticle } from 'services/article-service';
import { FETCH_ARTICLE_RESPONDED } from 'common/article-actions';
import { selectArticleName } from './article-selector';

export function fetchArticleAction() {
  return (dispatch, getState) =>
    fetchArticle(selectArticleName()(getState())).then(article => {
      dispatch({ type: FETCH_ARTICLE_RESPONDED, article });
    });
}
