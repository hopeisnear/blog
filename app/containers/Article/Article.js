import React from 'react';
import { Helmet } from 'react-helmet';
import ArticleDetails from 'components/ArticleDetails';

import './article-style.scss';

export default class Article extends React.Component {
  componentDidMount() {
    this.props.fetchArticle();
  }

  render() {
    return (
      <div className="article-details">
        <Helmet>
          <title>Feature Page</title>
          <meta
            name="description"
            content="Feature page of React.js Boilerplate application"
          />
        </Helmet>

        <div>
          {this.props.article && <ArticleDetails article={this.props.article} />}
        </div>
      </div>
    );
  }
}
