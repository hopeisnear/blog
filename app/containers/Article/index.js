import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import injectReducer from 'utils/injectReducer';

import { fetchArticleAction } from './article-actions';
import { selectArticle } from './article-selector';
import articleReducer from './article-reducer';
import Article from './Article';

const mapDispatchToProps = (dispatch) => ({
  fetchArticle: () => dispatch(fetchArticleAction({ id: 'test' }))
});

const mapStateToProps = createStructuredSelector({
  article: selectArticle()
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'article', reducer: articleReducer });

export default compose(withReducer, withConnect)(Article);
export { mapDispatchToProps };
