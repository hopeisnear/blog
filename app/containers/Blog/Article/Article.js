import React, { PureComponent } from 'react';
import { func, string, shape } from 'prop-types';
import { Helmet } from 'react-helmet';
import ArticleContent from 'components/ArticleContent';
import Comments from 'components/Comments';
import AddComment from 'containers/Blog/AddComment';

import './Article.scss';

export default class Article extends PureComponent {
  componentDidMount() {
    if (!this.props.article) {
      this.props.fetchArticle(this.props.articleName);
    }
  }

  render() {
    return (
      <div className="Article">
        <Helmet>
          <title>Article details</title>
          <meta
            name="description"
            content={`Article ${this.props.articleName} content`}
          />
        </Helmet>
        <div className="article__content">
          {this.props.article && <ArticleContent article={this.props.article} />}
        </div>
        {this.props.article && this.props.article.comments && <Comments comments={this.props.article.comments} />}
        <h3 className="article__reply_header">Leave a Reply</h3>
        <AddComment />
      </div>
    );
  }
}

Article.propTypes = {
  article: shape({}),
  articleName: string.isRequired,
  fetchArticle: func.isRequired
};

Article.defaultProps = {
  article: undefined
};
