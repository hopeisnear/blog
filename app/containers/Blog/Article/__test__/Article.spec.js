import React from 'react';
import { shallow } from 'enzyme';
import { spy, assert } from 'sinon';

import Article from '../Article';

describe('<Article />', () => {
  test('should render <Article />', () => {
    const articleComponent = shallow(<Article {...createProps()} />);

    expect(articleComponent).toMatchSnapshot();
  });

  test('should fetch new article on init', () => {
    const articleComponent = shallow(<Article {...createProps()} article={undefined} />);

    assert.calledWithExactly(articleComponent.instance().props.fetchArticle, 'Test article');
  });

  function createProps() {
    return {
      articleName: 'Test article',
      article: {
        comments: []
      },
      fetchArticle: spy()
    };
  }
});
