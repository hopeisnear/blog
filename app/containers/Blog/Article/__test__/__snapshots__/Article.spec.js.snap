// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render <Article /> 1`] = `
<div
  className="Article"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Article details
    </title>
    <meta
      content="Article Test article content"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="article__content"
  >
    <ArticleContent
      article={
        Object {
          "comments": Array [],
        }
      }
    />
  </div>
  <Comments
    comments={Array []}
  />
  <h3
    className="article__reply_header"
  >
    Leave a Reply
  </h3>
  <Connect(AddComment) />
</div>
`;
